framework/src/media/Decoder.cpp

```diff
@@ -67,17 +68,44 @@ size_t Decoder::pushData(unsigned char *buf, size_t size)
 	return 0;
 }
 
+/**
+ * @brief   Get decoded PCM sample frames
+ * @remarks
+ * @param   buf:  pointer to buffer for output
+ * @param   size: input/output parameter, in bytes.
+ *                input-size means the capability of 'buf', and also be the requested size for output.
+ *                output-size means the real output data size.
+ * @param   sampleRate:  Sample rate (Hz) of the output PCM samples
+ * @param   channels:  Number of channels of the output PCM samples
+ * @return  true  - output some data to 'buf', and output-size in the range of (0, input-size];
+ *          false - no output, and need push more audio data for decoding.
+ * @see
+ */
 bool Decoder::getFrame(unsigned char *buf, size_t *size, unsigned int *sampleRate, unsigned short *channels)
 {
 #ifdef CONFIG_AUDIO_CODEC
+	// Record pcm data fetched from audio decoder.
+	// Currently, pcm_data_t.samples point to the static buffer configured in Decoder::_configFunc(),
+	// so we can keep this pcm_data_t info before next pv_player_frame_decode() called.
+	static pcm_data_t pcm = {0};
+
+	// Get input buffer max size (it's also the requested size)
+	size_t max = *size;
+
+	// Check 2 bytes aligned
+	assert((max % 2) == 0);
+
+	// Reset output *size 0
+	*size = 0;
+
 	/*
 	 * Need to get the enough data to parse data format.
 	 */
 	if (mPlayer.audio_type == AUDIO_TYPE_UNKNOWN) {
-		mPlayer.audio_type = _get_audio_type(mPlayer.rbsp);
+		mPlayer.audio_type = pv_player_get_audio_type(&mPlayer);
```

Comment:
Could you please add sub message in commit message that we changed api to latest one from deprecated one?

Reply:
Added